frontend/services/upgrade.ts

```
@@ -206,9 +220,16 @@ export async function upgradeFrom_0_2_3(
     calldata: CallData.compile({ implementation: targetImplementationClassHash, calldata: upgradeCalldata }),
   };
 
-  const submitResult = await accountToUpgrade.execute([call]);
-  logger.log("upgrade to v0.4.0 transaction hash", submitResult.transaction_hash);
-  return submitResult.transaction_hash;
+  try {
+    const submitResult = await accountToUpgrade.execute([call]);
+    logger.log("upgrade to v0.4.0 transaction hash", submitResult.transaction_hash);
+    return submitResult.transaction_hash;
+  } catch (err) {
+    if (err instanceof Error && err.message.includes("exceed balance")) {
+      throw new Error("Not enough STRK to pay for the upgrade transaction");
```

i'd also include the original error to avoid hiding error information